### PR TITLE
DOC: Miscellaneous doc formatting fixes (part 4)

### PR DESCRIPTION
**Pull request by jhlegarreta**
- DOC: Add documentation files to toctree
- DOC: Enclose star-containing file extension text with double backticks
- DOC: Document method parameters
- STYLE: Remove emacs and vi editor mode lines
- DOC: Fix unexpected indentation errors related to references
- DOC: Indent properly references
- DOC: Use the appropriate syntax to link references
- DOC: Fix formatting of lists in method docstrings
- DOC: Use appropriate syntax for math equations
- DOC: Document appropriately the variable length and kw argument list
- DOC: Use the `func` directive to cross-reference methods
- DOC: Fix docstring indentation
- DOC: Enclose star-containing docstring text with double backticks
- DOC: Remove parameter default values from docstrings
- DOC: Document an optional parameter as such
- DOC: Fix miscellaneous typos
- DOC: Use double backticks for registration transform options in flow
- DOC: Add missing ending double backtick markup
- DOC: Do not leave empty line in method description docstring start
- DOC: Remove raw docstring marker where not necessary
- DOC: Add blank line before parameter block in method docstring
- DOC: Fix docstring indentation by removing unnecessary default value
- DOC: Add a blank line between method docstring parameter, return

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Before: 1186 warnings in:
https://github.com/dipy/dipy/actions/runs/10310154443/job/28541321104#step:5:2448

After: 1083 warnings in
https://github.com/dipy/dipy/actions/runs/10333627709/job/28606180924?pr=3309#step:5:2345

**Comment by jhlegarreta** [comment]
Build failures are unrelated to the changes in this patch set.

File: dipy/io/dpy.py [hunk 1]
```diff
@@ -25,11 +25,13 @@ def __init__(self, fname, *, mode="r", compression=0):
...
         Parameters
         ----------
-        fname : str, full filename
-        mode : 'r' read
-         'w' write
-         'r+' read and write only if file already exists
-        compression : 0 no compression to 9 maximum compression
+        fname : str
+            Full filename
+        mode : str
```
Comment (skoudoro) [inline dipy/io/dpy.py:30]: `str, optional` instead

File: dipy/io/dpy.py [hunk 2]
```diff
@@ -25,11 +25,13 @@ def __init__(self, fname, *, mode="r", compression=0):
...
+            Use 'r' to read, 'w' to write, and 'r+' to read and write (only if
+            file already exists).
+        compression : int
```
Comment (skoudoro) [inline dipy/io/dpy.py:33]: `int, optional` instead

**Review by skoudoro — COMMENTED** [review]
Thank you for this @jhlegarreta!

All good, see below 2 small comments. and then, it can be merged.

File: dipy/io/dpy.py (hunk 1 above)
Comment (jhlegarreta) [reply dipy/io/dpy.py]: Done in 437ce29.

File: dipy/io/dpy.py (hunk 2 above)
Comment (jhlegarreta) [reply dipy/io/dpy.py]: Done in 437ce29.

**Comment by jhlegarreta** [comment]
Build failures are unrelated.

**Review by skoudoro — APPROVED** [review]
All Good, merging

**Comment by skoudoro** [comment]
thank you @jhlegarreta

**Pull request merged**